### ISSUE TITLE
NLog Support for .NET Core

### ISSUE THREAD

**Issue by akatz0813** [aws/aws-logging-dotnet#8]
Do you support this?  Doesnt look like it since you arent targeting the NLog plugin for core.

**Comment by spati2** [comment]
NLog currently supports core in its beta versions. You can add the version 4.4.0-beta5 to your core application and use the logging library to log to Cloudwatch Logs

**Comment by akatz0813** [comment]
Sorry - but if I use this in my project.json:

` "dependencies": {
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.1"
    },
    "AWS.Logger.NLog": "1.1.0",
    "NLog": " 4.4.0-beta5"
  },`

Then I get: 

Package AWS.Logger.NLog 1.1.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package AWS.Logger.NLog 1.1.0 supports: net45 (.NETFramework,Version=v4.5)
    One or more packages are incompatible with .NETCoreApp,Version=v1.0.

**Comment by spati2** [comment]
could you post the entire project,json here?

**Comment by spati2** [comment]
I was able to create a netcore project and use the AWS.Logger.NLog library. Here is my project.json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },

  "dependencies": {
    "AWS.Logger.NLog": "1.1.0",
    "NLog": "4.4.0-beta5",
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.1"
    }
  },

  "frameworks": {
    "netcoreapp1.0": {
      "imports": "dnxcore50"
    }
  }
}

**Comment by akatz0813** [comment]
Can you post results from `dotnet restore -v Debug` ?  `dotnet restore -v Debug 1> restore.log 2>&1` will output it all to a file.  I'm curious if you have a source server that has a different release than nuget.  Also, make sure your building a project that can not link to the source code of this repo.  If there is a global.json in a parent directory that then contains both this test project and this repo, then it will not use nuget. In my experience, when nuget doesnt show that that your package targets .NET Standard 1.x, this error will appear.  See screenshots highlighting difference.

![image](https://cloud.githubusercontent.com/assets/7572738/25948102/31c618d8-3620-11e7-9f68-2c39f687eb02.png)


![image](https://cloud.githubusercontent.com/assets/7572738/25948089/24616a4e-3620-11e7-9039-9e34a15b0f8c.png)


My result using your project.json is:


[restore.txt](https://github.com/aws/aws-logging-dotnet/files/992875/restore.txt)

**Comment by spati2** [comment]
[restore.txt](https://github.com/aws/aws-logging-dotnet/files/993982/restore.txt)

@akatz0813 I have followed the instructions as you have specified and am posting the restore log. I have tested the nuget package with projects that target v4.5.2 and netcoreapp1.0

**Comment by akatz0813** [comment]
There's a chance you have some local packages here:  C:\AWSLogger\aws-logging-dotnet\Deployment 

Can you wipe that out and try again?  My guess is that you have a version locally that you haven't pushed to nuget.org

**Comment by spati2** [comment]
[restore.txt](https://github.com/aws/aws-logging-dotnet/files/994881/restore.txt)

Here you go.

**Comment by milla** [comment]
I have the same problem, and same question, I'm trying if I can deploy a netcore1.0 app to ECS, with AWS logger on CloudWatch, but also got the error.
`Restoring NuGet packages...
Package AWS.Logger.NLog 1.1.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package AWS.Logger.NLog 1.1.0 supports: net45 (.NETFramework,Version=v4.5)
One or more packages are incompatible with .NETCoreApp,Version=v1.0.
NuGet package restore failed.`

Windows 10, Visual Studio 2017, 
csproj file is as below,

`<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup Label="Globals">
    <SccProjectName>SAK</SccProjectName>
    <SccProvider>SAK</SccProvider>
    <SccAuxPath>SAK</SccAuxPath>
    <SccLocalPath>SAK</SccLocalPath>
  </PropertyGroup>

  <PropertyGroup>
    <TargetFramework>netcoreapp1.0</TargetFramework>
    <DockerComposeProjectPath>..\..\docker-compose.dcproj</DockerComposeProjectPath>
  </PropertyGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="AWS.Logger.NLog" Version="1.1.0" />
    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.0.0" />
    <PackageReference Include="Microsoft.AspNetCore" Version="1.0.3" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="1.0.3" />
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="1.0.1" />
  </ItemGroup>
  <ItemGroup>
    <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="1.0.0" />
  </ItemGroup>

</Project>
`

**Comment by spati2** [comment]
Hi @milla and @akatz0813 , when you are installing the package AWS.Logger.NLog and just that package alone, does dotnet try to install Nlog 4.4.9 package as a part of that installation?

**Comment by akatz0813** [comment]
As you can see in my restore.txt, "warn : Detected package downgrade: NLog from 4.4.9 to 4.4.0-beta5 "

**Comment by milla** [comment]
@spati2 quite confusing about the reference dependency now :( does this package supports .net core 1.0?

```
PM> Install-Package NLog -Pre
Restoring packages for ...
**Install-Package : Package AWS.Logger.NLog 1.1.0 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package AWS.Logger.NLog 1.1.0 
supports: net45 (.NETFramework,Version=v4.5)**
At line:1 char:1
+ Install-Package NLog -Pre
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand
 
Install-Package : One or more packages are incompatible with .NETCoreApp,Version=v1.0.
At line:1 char:1
+ Install-Package NLog -Pre
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand
 
Install-Package : Package restore failed. Rolling back package changes for 'OpenAPI.References.Web'.
At line:1 char:1
+ Install-Package NLog -Pre
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand
 
Time Elapsed: 00:00:03.6569427
```

**Comment by spati2** [comment]
Please use the latest Nuget package at  AWS.Logger.NLog 1.1.1 and let me know if the issue persists. Make sure that you have included the pre-release nuget package NLog 5.0.0-beta07

**Comment by akatz0813** [comment]
Works for me.
